extensions/amp-bind/amp-bind.md

```diff
@@ -66,425 +72,972 @@ The `amp-bind` component allows you to add custom stateful interactivity to your
+- The `<amp-img>` `src` points to a url.
+- Changeable elements have **bindings** that point to **expressions**.
+- The `[class]` attribute on the `<div>` is bound to the `theFood[currentMeal].style` **expression**.
+- The `[text]` attribute on the second `<p>` is bound to the `theFood[currentMeal].text` **expression**.
 
-<!-- Or change an image's src with the [src] binding. -->
-<amp-img
-  width="300"
-  height="200"
-  src="/img/dog.jpg"
-  [src]="myAnimals[currentAnimal].imageUrl"
->
-</amp-img>
+If a user clicks the "Set to sushi" button:
+
+1. The `tap` event trigger the `AMP.setState` action.
+1. The setState action turns `currentMeal` into a state and sets it to `sushi`.
+1. AMP evaluates **bindings** with **expressions** that contain the state `currentMeal`.
+1. `[class]="theFood[currentMeal].style` updates `class` to `redBorder`.
+1. `theFood[currentMeal].text` updates the inner text of the second `<p>` element to "Actually, I want to eat sushi.".
+
+Using `[class]="theFood[currentMeal].style"` as an example of **expression** syntax evaluation:
 
-<button on="tap:AMP.setState({currentAnimal: 'cat'})">Set to Cat</button>
+- `[class]` is the property to update
+- `theFood` is the id of the `<amp-state>` component.
+- `currentMeal` is the state name. In the case of `theFood` it will be `cupcakes` or `sushi`.
+- `style` is the **state variable**. It corresponds to the matching JSON key, and sets the bound property to that key's value.
+
+[/filter] <!-- formats="email" -->
+
+### `<amp-state>` specification
+
+[filter formats="websites, stories, ads"]
+An `amp-state` element may contain either a child `<script>` element **OR** a `src` attribute containing a CORS URL to a remote JSON endpoint, but not both.
+
+```html
+<amp-state id="myLocalState">
+  <script type="application/json">
+    {
+      "foo": "bar"
+    }
+  </script>
+</amp-state>
+
+<amp-state id="myRemoteState" src="https://data.com/articles.json"> </amp-state>
 ```
 
 [/filter] <!-- formats="websites, stories, ads" -->
 
 [filter formats="email"]
+An `amp-state` element must contain a child `<script>` element.
 
 ```html
-<!-- Store complex nested JSON data in <amp-state> elements. -->
-<amp-state id="myAnimals">
+<amp-state id="myLocalState">
   <script type="application/json">
     {
-      "dog": {
-        "imageUrl": "/img/dog.jpg",
-        "style": "greenBackground"
-      },
-      "cat": {
-        "imageUrl": "/img/cat.jpg",
-        "style": "redBackground"
+      "foo": "bar"
+    }
+  </script>
+</amp-state>
+```
+
+[/filter] <!-- formats="email" -->
+
+[filter formats="websites, stories, ads"]
+
+#### Attributes
+
+[filter formats="websites, stories, ads"]
+
+##### `src` (optional)
+
+The URL of the remote endpoint that will return the JSON and update this `amp-state`. This must be a CORS HTTP service. The `src` attribute allows all standard URL variable substitutions. See the [Substitutions Guide](../../spec/amp-var-substitutions.md) for more info.
```

Comment:
```suggestion
The URL of the remote endpoint that must return JSON, which is used to this `amp-state`. This must be a HTTP service with a proper CORS configuration for the page. The `src` attribute allows all standard URL variable substitutions. See the [Substitutions Guide](../../spec/amp-var-substitutions.md) for more info.
```

Reply:
Another nit here, but reinforcing these responses have to be JSON and correctly configured might help readers get on the path to success earlier.